browser/ipfs/ipfs_service_delegate_impl.h

```diff
@@ -0,0 +1,24 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_BROWSER_IPFS_IPFS_SERVICE_DELEGATE_IMPL_H_
+#define BRAVE_BROWSER_IPFS_IPFS_SERVICE_DELEGATE_IMPL_H_
+
+#include "brave/components/ipfs/browser/ipfs_service_delegate.h"
+
+namespace ipfs {
+
+class IpfsServiceDelegateImpl : public IpfsServiceDelegate {
+ public:
+  explicit IpfsServiceDelegateImpl(content::BrowserContext* context);
+  ~IpfsServiceDelegateImpl() override;
```

Comment:
there should be some boilerplate that deprecates copying
```
IpfsServiceDelegateImpl(const IpfsServiceDelegateImpl&) = delete;
IpfsServiceDelegateImpl& operator=(IpfsServiceDelegateImpl&) = delete;
```

Reply:
probably just `DISALLOW_COPY_AND_ASSIGN` is preferred.

Reply:
@bbondy Actually DISALLOW_* macros are deprecated, so we should not use that in new codes.